CMakeLists.txt

```diff
@@ -767,9 +767,10 @@ endif()
 set(CUDA_CXX_FLAGS "")
 
 if (LLAMA_CUBLAS)
-    set(CUDA_FLAGS ${CXX_FLAGS} -use_fast_math)
+    set(CUDA_FLAGS -use_fast_math)
+    list(APPEND CUDA_CXX_FLAGS ${CXX_FLAGS})
```

Comment:
I think this line can be removed entirely, the `CXX_FLAGS` is used for warnings, and that is already being passed to `nvcc` in the next block.
```suggestion
```

Reply:
Inside `build` directory: `cat CMakeFiles/ggml.dir/flags.make | grep Xcompiler`

Before removing line:

```
CUDA_FLAGS = -O3 -DNDEBUG -std=c++11 "--generate-code=arch=compute_52,code=[compute_52,sm_52]" "--generate-code=arch=compute_61,code=[compute_61,sm_61]" "--generate-code=arch=compute_70,code=[compute_70,sm_70]" -use_fast_math -Xcompiler "-Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wmissing-declarations -Wmissing-noreturn -Wno-pedantic -Wno-array-bounds -Wno-format-truncation -Wextra-semi -march=native"
```

After removing line:

```
CUDA_FLAGS = -O3 -DNDEBUG -std=c++11 "--generate-code=arch=compute_52,code=[compute_52,sm_52]" "--generate-code=arch=compute_61,code=[compute_61,sm_61]" "--generate-code=arch=compute_70,code=[compute_70,sm_70]" -use_fast_math -Xcompiler "-Wno-pedantic -Wno-array-bounds -Wno-format-truncation -Wextra-semi -march=native"
```

Can you double-check?

Reply:
Yes, it is still wrong with that change alone. There are a few inconsistencies in the way the CUDA host compiler flags are being set.